From e294ac988c69088e23ce091c09567c96baeafc19 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Fri, 1 Aug 2014 18:18:47 -0400
Subject: [tty] hvc: don't free hvc_console_setup after init

Message-id: <20140801141847.681c04bf@redhat.com>
Patchwork-id: 86941
O-Subject: [RHEL7.1 PATCH] drivers/tty/hvc: don't free hvc_console_setup after init
Bugzilla: 976490
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=976490
Brew-build: https://brewweb.devel.redhat.com/taskinfo?taskID=7776397
Upstream-status: merged

When 'console=hvc0' is specified to the kernel parameter in x86 KVM guest,
hvc console is setup within a kthread. However, that will cause SEGV
and the boot will fail when the driver is builtin to the kernel,
because currently hvc_console_setup() is annotated with '__init'. This
patch removes '__init' to boot the guest successfully with 'console=hvc0'.

Signed-off-by: Tomoki Sekiyama <tomoki.sekiyama@hds.com>
Cc: stable <stable@vger.kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 501fed45b7e8836ee9373f4d31e2d85e3db6103a)
Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/tty/hvc/hvc_console.c b/drivers/tty/hvc/hvc_console.c
index 9eba119..660f9f5 100644
--- a/drivers/tty/hvc/hvc_console.c
+++ b/drivers/tty/hvc/hvc_console.c
@@ -186,7 +186,7 @@ static struct tty_driver *hvc_console_device(struct console *c, int *index)
  return hvc_driver;
 }
 
-static int __init hvc_console_setup(struct console *co, char *options)
+static int hvc_console_setup(struct console *co, char *options)
 { 
  if (co->index < 0 || co->index >= MAX_NR_HVC_CONSOLES)
   return -ENODEV;
-- 
1.7.1